lib/companies_house_hub/models/filing_history.rb

```diff
@@ -42,14 +42,17 @@ def initialize(json = {})
       @transaction_id = json.dig(:transaction_id) # helps on barcode generation
       @barcode = json.dig(:barcode) || generate_barcode
       @description_values = json.dig(:description_values)
+      @company_number = company_number
     end
 
     def formatted_name
       FilingHistoryFormatter.new(self).formatted
     end
 
     def url(format = 'pdf')
-      file_path = @links[:self]
+      file_path = @links[:self] || build_file_path
+
+      return unless file_path
```

Comment:
Not sure about this, whether it should return `""` or `nil` when it can't build url, 
nil seems clearer, while empty string will keep contract of always returning string, even if it's invalid